src/js/port_handler.js

```diff
@@ -242,12 +242,15 @@ PortHandler.selectPort = function(ports) {
     for (let i = 0; i < ports.length; i++) {
         const portName = ports[i].displayName;
         if (portName) {
+            const pathSelect = ports[i].path;
             if (portName.includes('STM') || portName.includes('CP210')) {
-                const pathSelect = ports[i].path;
                 if (OS === 'Windows' || (OS !== 'Windows' && pathSelect.includes('tty'))) {
                     this.portPickerElement.val(pathSelect);
                     console.log(`Porthandler detected device ${portName} on port: ${pathSelect}`);
                 }
+            } else if (portName.includes('tty') && portName.includes("usb")) {
+                this.portPickerElement.val(pathSelect);
+                console.log(`Porthandler detected legacy device ${portName} on port: ${pathSelect}`);
```

Comment:
I don't know if we need all this if-else only to show a different log message.

I think is better to unify all of them. We do some extra verification, but I think the overhead is acceptable if the code is better enough. Something like:
```javascript
const boolean isWindows = (OS === 'Windows');
const boolean isTty = pathSelect.includes('tty');
const boolean driverRecognized = portName.includes('STM') || portName.includes('CP210');
const boolean legacyDriverRecognized = portName.includes('usb');
if ((driverRecognized && (isWindows || isTty)) || (legacyDriverRecognized && isTty) {
```
Maybe we can remove the de isWindows and isTty conditions, but you have tested it more.

Reply:
Looks very nice. First implementation had no duplication. But the legacy check introduced some. This seems a clever alternative.